types/prosemirror-view/index.d.ts

```diff
@@ -568,11 +568,19 @@ export interface EditorProps<ThisT = unknown, S extends Schema = any> {
    */
   nodeViews?: {
     [name: string]: (
-      node: ProsemirrorNode<S>,
-      view: EditorView<S>,
-      getPos: (() => number) | boolean,
-      decorations: Decoration[]
-    ) => NodeView<S>;
+      ((
+        node: ProsemirrorNode<S>,
+        view: EditorView<S>,
+        getPos: (() => number),
+        decorations: Decoration[])
+        => NodeView<S>) |
+      ((
+        mark: Mark<S>,
```

Comment:
Can you help me check if the union type is backwards compatible with lower versions? I'm actually not sure if it's the union type in `index.d.ts`, or the way `$ExpectError` should placed within `prosemirror-view-tests.ts`.

The reason why I thought it was 3.8 is because when I ran `npm run lint prosemirror-view`, I got the following
```
ERROR: 118:3  expect  Compile error in typescript@3.7 but not in typescript@3.8.
Fix with a comment '// Minimum TypeScript Version: 3.8' just under the header.
Type '{ markview: (mark: model.Node<any>, view: view.EditorView<any>, getPos: boolean, decorations: view.Decoration<{ [key: string]: any; }>[]) => any; nodeview: (node: model.Mark<any>, view: view.EditorView<...>, getPos: () => number, decorations: view.Decoration<...>[]) => any; }' is not assignable to type '{ [name: string]: ((node: ProsemirrorNode<any>, view: EditorView<any>, getPos: () => number, decorations: Decoration<{ [key: string]: any; }>[]) => NodeView<any>) | ((mark: Mark<...>, view: EditorView<...>, getPos: boolean, decorations: Decoration<...>[]) => MarkView<...>); }'.
  Property 'markview' is incompatible with index signature.
    Type '(mark: model.Node<any>, view: view.EditorView<any>, getPos: boolean, decorations: view.Decoration<{ [key: string]: any; }>[]) => any' is not assignable to type '((node: ProsemirrorNode<any>, view: EditorView<any>, getPos: () => number, decorations: Decoration<{ [key: string]: any; }>[]) => NodeView<any>) | ((mark: Mark<...>, view: EditorView<...>, getPos: boolean, decorations: Decoration<...>[]) => MarkView<...>)'.
      Type '(mark: model.Node<any>, view: view.EditorView<any>, getPos: boolean, decorations: view.Decoration<{ [key: string]: any; }>[]) => any' is not assignable to type '(node: ProsemirrorNode<any>, view: EditorView<any>, getPos: () => number, decorations: Decoration<{ [key: string]: any; }>[]) => NodeView<any>'.
        Types of parameters 'getPos' and 'getPos' are incompatible.
          Type '() => number' is not assignable to type 'boolean'.
```

but I can't pinpoint in the 3.8 release notes (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html) any mention of union type.

Reply:
I believe the normal pattern would be to put the _older_ version in a folder and keep the newest version in the root.  At least, I assume that's why dtslint complains about `typesVersions` specification.

Reply:
There were a bunch of union changes in the 3.8/3.9 timeframe, but I didn't track them closely enough to be able to tell you which affected your change.

Reply:
Hm, does the latest commit (https://github.com/BrianHung/DefinitelyTyped/commit/d9d4219e1631f7002f7f45111e32574563cd31c9) "put the older version in a folder and keep the newest version in the root"? 

The only difference between this commit and the one we're commenting on is that I added backwards-compatible tests for 3.1 and 3.8; I did not change the types themselves.

`npm run lint prosemirror-view` and `npm test` run without error so it seems fine? I'm still fairly new to testing with Typescript.

Reply:
It would be easier for me to review if you pushed changes here (which I assume you weren't doing because the PR was closed).

Reply:
I didn't see a new commit to the PR, so I looked at the branch in your fork.  It looks like there's still a ts3.8 folder, so it's not in the structure I expected.  I thought the code for the newest version of TypeScript would go in the root directory (`types/prosemirror-view`).